### ISSUE TITLE
Bug Report: Issue with `getTableNames` in Yii2 Schema Handling on sqlite3 database

### ISSUE THREAD

**Issue by santilin** [yiisoft/yii2#20262]
**Summary:**
There is a bug in the `findTableNames` method of the `Schema` class in Yii2 that affects the retrieval of table names when a schema is specified. The current implementation does not correctly handle the schema prefix, leading to incorrect SQL queries for SQLite databases.

**Environment:**
- Yii2 Framework Version: develop
- Database: SQLite

**Description:**
The method `findTableNames` is designed to return distinct table names from the SQLite database. However, when a schema is provided, the SQL query constructed does not correctly reference the `sqlite_master` table. The modification proposed in the code snippet aims to prepend the schema to the `sqlite_master` reference, but it fails to account for the correct SQL syntax.

**Code Snippet:**
```php
protected function findTableNames($schema = '')
{
    if ($schema != '') {
        $schema .= '.';
    }
    $sql = "SELECT DISTINCT tbl_name FROM {$schema}sqlite_master WHERE tbl_name<>'sqlite_sequence' ORDER BY tbl_name";
    return $this->db->createCommand($sql)->queryColumn();
}
```

**Expected Behavior:**
When calling `findTableNames('my_schema')`, the expected SQL should be:
```sql
SELECT DISTINCT tbl_name FROM my_schema.sqlite_master WHERE tbl_name<>'sqlite_sequence' ORDER BY tbl_name;
```
This query should return all table names under the specified schema.

**Actual Behavior:**
The current implementation results in an SQL error because `sqlite_master` does not support schema prefixes in SQLite. The correct way to access the master table does not involve a schema prefix, as it is always accessed directly without any schema context.

**Steps to Reproduce:**
1. Set up a Yii2 application connected to an SQLite database.
2. Create a schema and several tables within that schema.
3. Call the `findTableNames` method with the schema name as a parameter.
4. Observe that an SQL error occurs or that no tables are returned.

**Proposed Fix:**
Modify the SQL query in the `findTableNames` method to remove any schema prefix when querying `sqlite_master`. The corrected code should look like this:
```php
protected function findTableNames($schema = '')
{
    $sql = "SELECT DISTINCT tbl_name FROM sqlite_master WHERE tbl_name<>'sqlite_sequence' ORDER BY tbl_name";
    return $this->db->createCommand($sql)->queryColumn();
}
```

**Additional Information:**
This issue has been noted in various discussions around Yii2's handling of schemas and table names, particularly for SQLite databases. It is crucial to ensure that methods like `findTableNames` behave consistently across different database types and do not introduce errors due to incorrect SQL syntax.

**References:**
- Yii2 API Documentation on Schema [1]
- Community discussions on related issues [4][5][6]

Citations:
[1] https://www.yiiframework.com/doc/api/2.0/yii-db-schema
[2] https://www.yiiframework.com/doc/api/2.0/yii-db-mysql-schema
[3] https://www.yiiframework.com/doc/api/2.0/yii-db-sqlite-schema
[4] https://forum.yiiframework.com/t/cdbschema-findtablenames-not-part-of-schemacachingduration/73388
[5] https://github.com/yiisoft/yii2/issues/8096
[6] https://github.com/yiisoft/yii/issues/2299
[7] https://forum.yiiframework.com/t/table-name-issue-in-oracle-db-and-yii2/79566
[8] https://www.yiiframework.com/doc/api/2.0/yii-db-oci-schema

**Comment by samdark** [comment]
@Tigrov would you please take a look? Is it the same in yiisoft/db?

**Comment by Tigrov** [comment]
Current Yii SQLite driver does not support scheme names. 

From the [documentation](https://www.sqlite.org/lang_createtable.html)
> If a schema-name is specified, it must be either "main", "temp", or the name of an [attached database](https://www.sqlite.org/lang_attach.html).

It makes sense to add support schema names. In Yii3 the same.

**Comment by santilin** [comment]
In my yii2-sqlite3-full-support proyect I have added support for schemes replacing [yii\db\sqlite\Scheme.php](https://github.com/santilin/yii2-sqlite3-full-support/blob/master/src/db/sqlite/Schema.php) with my own version. I have been using it for a while and it works well, thogh it may not be complete. 

It would be great to include this onto yii2.

**Comment by Tigrov** [comment]
@santilin Could you make a PR for yii2 to support scheme names?

**Comment by santilin** [comment]
> @santilin Could you make a PR for yii2 to support scheme names?

Yes, but firstly, please, point me to a guide on correct formatting of the source code so that we don't get errors on th PR and another one for making a PR to yii2
